SPHINXsys/src/shared/particles/base_particles.cpp

```diff
@@ -233,6 +233,14 @@ namespace SPH
 	void BaseParticles::writePltFileHeader(std::ofstream& output_file)
 	{
 		output_file << " VARIABLES = \"x\",\"y\",\"z\",\"ID\"";
+		
+		for (size_t l = 0; l != variables_to_write_[indexInteger].size(); ++l) 
```

Comment:
is it better to make this function less relying on class members? indexInteger/indexVector/variables_to_write_/all_particle_data_  are function input?

This way it is easier to write test for the function and code might be easier to manage.

Reply:
Thanks for the comment ans I will update the suggested change in the next pull request.